### DISCUSSION TITLE
How to set maxSessions for webdriverio/selenium-standalone?

### DISCUSSION THREAD

**Discussion by GeorgeCodreanu** [webdriverio/webdriverio#10357] [Q&A]
Whenever I start my wdio tests, they connect run on selenium-standalone server which also has maxSessions set based on the number of CPU cores, for ex:
- on local machine its 10
- on azure agents its 2

How can I edit my config, so that it uses 5 for example regardless of CPU cores.

```
    services: [
        [
            'selenium-standalone',
            {
                logPath: 'logs',
                installArgs: { drivers }, // drivers to install
                args: { drivers }, // drivers to use
            },
        ],
    ],

```

The property `maxInstances` does not change it.

I want to run this on Azure agent and not have just 2 max concurrency.

Also asked here:
https://stackoverflow.com/questions/76199488/how-to-set-maxsessions-for-webdriverio-selenium-standalone

**Comment by christian-bromann** [comment]
> The property `maxInstances` does not change it.

What do you mean by that?

**Comment by GeorgeCodreanu** [comment]
@christian-bromann for instance i set 
`    maxInstances: 2,
`

and while the queue size will always be 2, the max concurrency will always be 10, as per this /status response from the grid:

```"value": {
    "ready": true,
    "message": "Selenium Grid ready.",
    "nodes": [
      {
        "id": "ab18ea85-2a14-4fdc-be30-dc3e4ec0715e",
        "uri": "http:\u002f\u002f10.1.9.233:4444",
        "maxSessions": 10,
        "osInfo": {
          "arch": "aarch64",
          "name": "Mac OS X",
          "version": "13.3.1"
        },
        "heartbeatPeriod": 60000,
        "availability": "UP",
        "version": "4.4.0 (revision e5c75ed026a)",
        "slots":  //rest of response
```

as you can see, maxSessions is still 10.

**Reply by christian-bromann** [reply]
`maxSessions` is a grid option and can be potentially set using the service [`args`](https://webdriver.io/docs/selenium-standalone-service/#args) option.
